lib/activerecord/delay_touching.rb

```diff
@@ -6,9 +6,9 @@ module DelayTouching
     extend ActiveSupport::Concern
 
     # Override ActiveRecord::Base#touch.
-    def touch(name = nil)
+    def touch(*names)
       if self.class.delay_touching? && !try(:no_touching?)
-        DelayTouching.add_record(self, name)
+        DelayTouching.add_record(self, names)
```

Comment:
Yep. I was missing a splat in the touch method that alleviates the need to flatten the array. I can greatly simplify the logic based on this. 

there are also a couple of minor tweaks that I plan to do. I'm going to close this PR and resubmit through a separate branch so that I can squash the changes.